packages/react/src/components/UIShell/Header.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import cx from 'classnames';
+import React, { ReactNode } from 'react';
+import { usePrefix } from '../../internal/usePrefix';
+
+interface AriaLabelPropType {
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+}
+
+interface HeaderProps extends AriaLabelPropType {
+  className?: string;
+  children?: ReactNode;
+}
+
+const Header: React.FC<HeaderProps> = ({
+  className: customClassName,
+  children,
+  ...rest
+}) => {
+  if (!rest['aria-label'] && !rest['aria-labelledby']) {
+    throw new Error(
+      'You must provide either an aria-label or aria-labelledby prop to the Header component.'
+    );
+  }
+  const prefix = usePrefix();
+  const className = cx(`${prefix}--header`, customClassName);
+
+  return (
+    <header {...rest} className={className}>
+      {children}
+    </header>
+  );
+};
+
```

Comment:
We should keep the propTypes for documentation reasons in storybook and for JavaScript only also.

```suggestion
Header.propTypes = {
  /**
   * Required props for the accessibility label of the header
   */
  ...AriaLabelPropType,

  /**
   * Optionally provide a custom class name that is applied to the underlying <header>
   */
  className: PropTypes.string,
};
```

Reply:
oops! think i read that at totally forgot!
committing the proptypes now!